objects/69ead2/contained/DUESETUP/script.lua

```diff
@@ -0,0 +1,643 @@
+    updateEnergyDisplay(card.guid)
+    updateSave()
+end
+
+function clearAll(card)
+    clearCardData(card.guid)
+end
+
+function clearCardData(guid)
+    clearCardButtons(guid)
+    impendTable[guid] = nil
+    local card = getObjectFromGUID(guid)
+    if card then
+        card.hide_when_face_down = true -- if card is in bag then oh well it can stay that way it really won't matter
+    end
+    updateSave()
+end
+
+function clearCardButtons(guid)
+    local card = getObjectFromGUID(guid)
+    if card then
+        for _,func in pairs({"energyDisplay", "overridePlus", "overrideMinus", "clearAll", "g3Minus", "g3Plus"}) do -- add to if needed
+            local ind = getButtonIndex(card, func)
+            if ind then
+                card.removeButton(ind)
+            end
+        end
+    end
+end
+
+--------------------------------------------
+-- card buttons for g3 and related functions
+--------------------------------------------
+
+function createAllG3Buttons()
+    if g3Selected then
+        for _,tbl in pairs(impendTable) do
+            card = getObjectFromGUID(tbl.guid)
+            if card then
+               createG3Buttons(card)
+            end
+        end
+    end
+end
+
+function createG3Buttons(card)
+    local tbl = impendTable[card.guid]
+    if tbl.turnSelected ~= Global.getVar("turn") and getButtonIndex(card,"g3Plus") == nil then
+        card.createButton({
+            click_function = "g3Plus",
+            function_owner = self,
+            label = "+",
+            position = {-0.55,aboveCard,-0.8},
+            width = 120,
+            height = 120,
+            font_size = 150,
+            color = {1,1,1,1},
+            tooltip = "Add 1 impended energy using the third growth option",
+        })
+
+        card.createButton({
+            click_function = "g3Minus",
+            function_owner = self,
+            label = "-",
+            position = {-0.55,aboveCard,-0.4},
+            width = 120,
+            height = 120,
+            font_size = 150,
+            color = {1,1,1,1},
+            tooltip = "Subtract 1 impended energy using the third growth option",
+        })
+    end
+end
+
+function removeAllG3Buttons()
+    for _,tbl in pairs(impendTable) do
+        card = getObjectFromGUID(tbl.guid)
+        if card then
+            removeG3Buttons(card)
+        end
+    end
+end
+
+function removeG3Buttons(card)
+    local tbl = impendTable[card.guid]
+    if tbl.turnSelected ~= Global.getVar("turn") and getButtonIndex(card,"g3Plus") ~= nil then
+        card.removeButton(getButtonIndex(card,"g3Plus"))
+        card.removeButton(getButtonIndex(card,"g3Minus"))
+        impendTable[tbl.guid].energy[3] = 0
+        updateEnergyDisplay(tbl.guid)
+    end
+end
+
+function updateG3Buttons(guid)
+    local card = getObjectFromGUID(guid)
+    if card then
+        local plus_color = impendTable[guid].energy[3] == 1 and {1,1,0,1} or {1,1,1,1}
```

Comment:
When G3 is selected, a "+" and a "-" button appear on the cards that have been impended on a previous turn. This line is to give a yellow coloration to the "+" button if it has been selected on that given card (which is only true when `impendTable[guid].energy[3] == 1`) and otherwise left white as shown in the image below. Similar code is done for the "-" button just below. This can be changed to be written as an if statement if preferable, I was just compactifying this bit as it just for aesthetics.

![20240409160539_1](https://github.com/spirit-island/spirit-island-mod/assets/4542823/84cc85e8-9fbe-4df7-b0dc-0a02a71658f4)

Reply:
ah so the idea is that `{1,1,0,1}` and `{1,1,1,1}` always evaluate to true, so depending upon the first part being true or false then the other ones would be chosen. it's basically just a ternary operator

Reply:
Yeah, that's unfortunately the only way people to do a ternary operator in Lua. I hate the notation personally (and it has some weird edge cases that can cause it to fail), so I've been avoiding it and just writing if-elses.